### ISSUE TITLE
Not view Juju Gui on a VPN connection

### ISSUE THREAD

**Issue by richardsith** [juju/juju-gui#3083]
I have an issue with juju, if I try to view its gui on a VPN connection the task rests in pending on "Requesting code..." as shown the picture

![vguk6](https://user-images.githubusercontent.com/9977215/28289255-369145ca-6b42-11e7-9764-78f1c8f955da.png)

![screen shot 2017-07-17 at 14 18 59](https://user-images.githubusercontent.com/9977215/28289411-c9a651ca-6b42-11e7-8ee9-c010b5b24448.png)

while I have not any problem with Landscape, Openstack or MAAS.

![wp72h](https://user-images.githubusercontent.com/9977215/28289288-5447960a-6b42-11e7-8efa-9f3a53ccf7a2.png)

**Comment by hatched** [comment]
@richardsith it looks like it's not able to access the Charmstore. On the same VPN can you access the following URL?

https://api.jujucharms.com/charmstore/v5/bundle/canonical-kubernetes/archive/bundle.yaml

**Comment by richardsith** [comment]
no, but Juju gui needs to work with a external connection?

**Comment by hatched** [comment]
The GUI needs to be able to access the charmstore to be able to display the charm data including the icons. However we should make it fail more gracefully here and at least be able to display the model with as much information as we can. I've created a new issue to resolve the GUI not loading #3086, you can subscribe to that issue to track when this should be resolved.

As a workaround the GUI will need access to api.jujucharms.com.

**Comment by richardsith** [comment]
from VM where installed Juju Gui the connection is correct.  Look the picture.

![screen shot 2017-07-21 at 23 13 16](https://user-images.githubusercontent.com/9977215/28494342-d9eb2834-6f2b-11e7-83db-bbecd6f17c4d.png)

**Comment by hatched** [comment]
It looks like you're trying to fetch the yaml file from an ubuntu machine, not the browser in which the GUI is loading. Are you able to view that yaml file from that link in the browser you're trying to open the GUI in?

**Comment by hatched** [comment]
@richardsith I've just landed a fix which will be in the next release which allows these requests to fail gracefully and still allow you to access the GUI without having access to the charmstore. Of course you won't be able to deploy anything but local charms when this lands as it cannot communicate with the charmstore.

**Comment by richardsith** [comment]
@hatched ok perfect. thanks. waiting that I try to modify the rules on my Fw to permit to Juju of working also in a VPN connection and access to charmstore.
